singleton.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zap
+
+import "sync"
+
+var (
+	singletonLogger *logger
+	initOnce        sync.Once
+	configLock      sync.Mutex
+)
+
+// Standard returns the standard, singleton logger. Although this behavior is discouraged in many settings
```

Comment:
The `StandardLogger` in the `zwrap` package refers to an interface that the standard library logger implements, while this one refers to a singleton implementing the `zap.Logger` interface. Having the same name reused for very different things sounds confusing.

We could rename the logger in `zwrap`, or we could do what log15 does, and call it `Root`:
https://godoc.org/github.com/inconshreveable/log15#Root

I'm also noticing that a lot of logging libraries don't expose a singleton at all. Even the standard library exposes functions for the singleton logger, but not the actual instance.